### PR TITLE
#1066 - Review documentation

### LINKED ISSUE #1066: Review documentation
We have a lot of documentation and I think we should stick with only the relevant one...

//cc @comicrelief/drupal-team

### PR DESCRIPTION
**Pull request by Saphyel**
Fixes #1066

## Changes proposed in this PR

- [x] Update release docs
- [ ] Update [testing docs](https://github.com/comicrelief/campaign/blob/develop/docs/testing.md), so you can run behat tests locally 

## Questions
* Do we need `feature-branching.md` ? and `naming-convention.md` ? (Or this should be part of `CONTRIBUTING.md` ?) and also maybe we should redirect to => https://github.com/comicrelief/engineering-handbook/ ?
* Do we need `plat.sh.md `? I think plat.sh has already good documentation
* Do we need `testing.md` ? it feels very outdated
* Do we need `troubleshooting.md` ?
* `development.md` should be part of the `install.md` ? feels like the same doc
* What is the difference between `README.md` and the `docs/index.md`? looks the same
* Rules_of_the_road.md it's a cool name, but it's nor very descriptive about what is

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Think about contributing? That's great, but please do read our [contributing gui
 * [Blog post: building rednoseday.com on Drupal 8](https://technology.comicrelief.com/2016/12/21/building-rednoseday-com-on-drupal-8/)
 * [Video DrupalCon Baltimore 2017: building reusable websites on Drupal 8](https://technology.comicrelief.com/2017/05/12/drupalcon-baltimore-talk-slides/)
 * [Blog post: Why we made our platform product open-source](https://technology.comicrelief.com/2017/07/07/why-we-made-our-platform-product-open-source/)
-* [Release process](docs/release.md)
+* [Release process](docs/releases.md)
```
Comment (pvhee) [inline README.md]: Add in platform.sh docs around preview branches

Comment (pvhee) [reply README.md]: Add in a link to the testing docs and make sure these docs are up to date

File: docs/platformsh.md [hunk 2]
```diff
@@ -1,53 +0,0 @@
-## Platform.sh
-
-We use [platform.sh](http://platform.sh) for deploying Pull Requests to environments.
```
Comment (pvhee) [inline docs/platformsh.md]: I wouldn't get rid of these docs just yet, these might be helpful to help us debug issues with the environment

Comment (Saphyel) [reply docs/platformsh.md]: @pvhee plat.sh has already a good documentation why do we need this? also they keep it up to date

I want to keep only relevant documentation

File: README.md (hunk 1 above)
Comment (Saphyel) [reply README.md]: Why do we need a testing docs? Nowadays run the tests are easy and if you follow the official documentation you won't have any issue (and before our tests require some conditions, and they were in strange location, etc...)


Maybe we should mention that we use behat for our tests and a link to the documentation?

Comment (pvhee) [reply README.md]: @Saphyel where can I find how to run the tests? Don't think we've got any documentation for that

File: docs/platformsh.md (hunk 2 above)
Comment (pvhee) [reply docs/platformsh.md]: @Saphyel ok, but let's then at least link off to plat.sh docs, and explain why we need / use feature branches

File: README.md (hunk 1 above)
Comment (Saphyel) [reply README.md]: We use behat... any question about behat or how to use it should be in behat not in our repo...

File: docs/platformsh.md (hunk 2 above)
Comment (Saphyel) [reply docs/platformsh.md]: That maybe part of CONTRIBUTING?

**Pull request merged**